go.mod

```diff
@@ -1,8 +1,8 @@
 module github.com/prometheus/prometheus
 
-go 1.21.0
+go 1.22.0
 
-toolchain go1.22.5
+toolchain go1.23.0
```

Comment:
Now that I'm thinking about it, with toolchain, `Go tests with previous Go version` are also using the latest version now, I think they need a `GOTOOLCHAIN=local`

https://github.com/prometheus/prometheus/actions/runs/10471451070/job/28998762258?pr=14697#step:4:5

Reply:
Done